### PR TITLE
[core] Fix comments and a corner case in #32302

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?

This is a corner case where buffer could be 0 and a comments needs to be fixed in the previous PR.
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/rpc/grpc_server.cc [hunk 1]
```diff
@@ -148,7 +149,7 @@ void GrpcServer::Run() {
       if (entry->GetMaxActiveRPCs() != -1) {
         buffer_size = entry->GetMaxActiveRPCs();
       }
-      for (int j = 0; j < (buffer_size / num_threads_); j++) {
+      for (int j = 0; j < std::min(1, buffer_size / num_threads_); j++) {
```
Comment (scv119) [inline src/ray/rpc/grpc_server.cc:152]: should it be `std::max` ?

Comment (iycheng) [reply src/ray/rpc/grpc_server.cc:152]: oops

**Review by scv119 — APPROVED** [review]

File: src/ray/rpc/grpc_server.cc [hunk 2]
```diff
@@ -40,6 +40,7 @@ GrpcServer::GrpcServer(std::string name,
       is_closed_(true),
       num_threads_(num_threads),
       keepalive_time_ms_(keepalive_time_ms) {
+  RAY_CHECK(num_threads_ > 0) << "Num of threads in gRPC must be greater than 0";
```
Comment (cadedaniel) [inline src/ray/rpc/grpc_server.cc:43]: seems like a good case for an unsigned type 😄

**Review by cadedaniel — APPROVED** [review]

File: src/ray/rpc/grpc_server.cc (hunk 2 above)
Comment (iycheng) [reply src/ray/rpc/grpc_server.cc:43]: Yes, but potentially it still could be 0 if it's unsigned type.

**Pull request merged**